config/esp32/components/chip/Kconfig

```diff
@@ -652,20 +652,30 @@ menu "CHIP Device Layer"
 
                 This option is for testing only and should not be enabled in production releases.
 
-        config USE_TEST_PAIRING_CODE
-            string "Use Test Pairing Code"
-            default "CHIPUS"
+        config USE_TEST_SETUP_PIN_CODE
+            string "Use Test Setup Pin Code"
+            range 0 99999999
```

Comment:
Sorry If I'm confused here. By setup pin code you mean the manual pairing code ?  If that's correct the renaming is a bit confusing since the manual pairing code contains a field named.. pin code, that is called in the codebase setUpPinCode (https://github.com/project-chip/connectedhomeip/blob/master/src/setup_payload/ManualSetupPayloadGenerator.cpp#L37)

Reply:
Setup Pin Code is for `setUpPinCode` in codebase, the 27bit integer.

Reply:
Good. I'm less confused then !

In this case, the valid range is between 0 and 134217727 (2^27-1) and 0x0 is normally used to specify that the device has already been paired (for device that supports dynamic generation of QRCode).

And the comments above says that this string is expected to end with a valid Verhoeff check character. But this part of the comment refers to the whole Pairing Code, not the setup pin code afaict.

Reply:
Oh, I forget to remove that, thanks.

I remembered that the spec said the pin code printed on the device is a 8 digit number, thus 99999999 requires 27 bits, although 27 bits can represent 1.3e9 in fact. I guess device with screen which can display a QR code should also follow this limit.

As for the 0x0, since 0x0 will never be a valid pin code (since it indicate the device is already paired in some case), I think it is also ok if 0x0 means not to use the default value.

However, I wonder if using 0x0 to disable default discriminator is a good choice since discriminator is just a 12bit random number.